pubsubs/registry.py

```
@@ -0,0 +1,57 @@
+""" Registry controlling access to pubsubs."""
+import yaml
+
+from pubsubs.base import MessageQueue
+
+
+class Registry:
+    def __init__(self):
+        self._registry = {}
+
+    def new(self, name, backend, **kwargs):
+        """ Register concrete instance."""
+        return self.register(
+            MessageQueue.for_backend(backend)(name, registry=self, **kwargs), name=name
+        )
+
+    def register(self, message_queue, name):
+        """ Register concrete message queue instance under a name."""
+        name = message_queue.name if not name else name
+
+        if name in self._registry:
+            raise ValueError(f"PubSub {name} exists.")
```

This should raise an internal exception.